cypress/e2e/checkin/check-in-multipiece-items.cy.js

```diff
@@ -0,0 +1,150 @@
+import MultipieceCheckIn from '../../support/fragments/checkin/modals/multipieceCheckIn';
+import CheckInActions from '../../support/fragments/check-in-actions/checkInActions';
+import Users from '../../support/fragments/users/users';
+import UserEdit from '../../support/fragments/users/userEdit';
+import InventoryInstance from '../../support/fragments/inventory/inventoryInstance';
+import DevTeams from '../../support/dictionary/devTeams';
+import { ITEM_STATUS_NAMES } from '../../support/constants';
+import Location from '../../support/fragments/settings/tenant/locations/newLocation';
+
+describe('Check In', () => {
+  let user = {};
+  let servicePoint;
+  let materialTypeName;
+  let testInstanceIds;
+  const instanceTitle = `autotest_instance_title_${getRandomPostfix()}`;
+  const testItems = [];
+  const defaultDescription = `autotest_description_${getRandomPostfix()}`;
+  let defaultLocation;
+
+  beforeEach(() => {
+    cy.createTempUser([permissions.checkinAll.gui])
+      .then((userProperties) => {
+        user = userProperties;
+        servicePoint = NewServicePoint.getDefaultServicePoint();
+        ServicePoints.createViaApi(servicePoint);
+        defaultLocation = Location.getDefaultLocation(servicePoint.id);
+        Location.createViaApi(defaultLocation);
+        UserEdit.addServicePointViaApi(servicePoint.id, user.userId, servicePoint.id);
+      })
+      .then(() => {
+        cy.login(user.username, user.password);
+      });
+    cy.getAdminToken()
+      .then(() => {
+        cy.getLoanTypes({ limit: 1 });
+        cy.getMaterialTypes({ limit: 1 }).then(({ id, name }) => {
+          materialTypeName = { id, name };
+        });
+        cy.getHoldingTypes({ limit: 2 });
+        cy.getInstanceTypes({ limit: 1 });
+      })
+      .then(() => {
+        const getTestItem = (numberOfPieces, hasDescription, hasMissingPieces) => {
+          const defaultItem = {
+            barcode: Helper.getRandomBarcode(),
+            status: { name: ITEM_STATUS_NAMES.AVAILABLE },
+            permanentLoanType: { id: Cypress.env('loanTypes')[0].id },
+            materialType: { id: materialTypeName.id },
+            materialTypeName: materialTypeName.name,
+            instanceTitle,
+          };
+          if (numberOfPieces) {
+            defaultItem.numberOfPieces = numberOfPieces;
+          }
+          if (hasDescription) {
+            defaultItem.descriptionOfPieces = defaultDescription;
+          }
+          if (hasMissingPieces) {
+            defaultItem.numberOfMissingPieces = 2;
+            defaultItem.missingPieces = defaultDescription;
+          }
+          return defaultItem;
+        };
+        testItems.push(getTestItem(1, false, false));
+        testItems.push(getTestItem(3, true, false));
+        testItems.push(getTestItem(2, true, true));
+        testItems.push(getTestItem(0, false, true));
+        InventoryInstances.createFolioInstanceViaApi({
```

Comment:
There are new implemenation of methods for generation and creation of Instance objects:
```JS
InventoryInstances.generateFolioInstances()
InventoryInstances.createFolioInstancesViaApi()
```

Could you please reuse them.

You can find examples of usages in the following test:
cypress/e2e/circulation-log/loan-periods.cy.js

Reply:
Done